x-pack/filebeat/input/azureblobstorage/input.go

```diff
@@ -85,25 +91,42 @@ func tryOverrideOrDefault(cfg config, c container) container {
 		}
 		c.MaxWorkers = &maxWorkers
 	}
-
 	if c.Poll == nil {
 		var poll bool
 		if cfg.Poll != nil {
 			poll = *cfg.Poll
 		}
 		c.Poll = &poll
 	}
-
 	if c.PollInterval == nil {
 		interval := time.Second * 300
 		if cfg.PollInterval != nil {
 			interval = *cfg.PollInterval
 		}
 		c.PollInterval = &interval
 	}
+	if c.TimeStampEpoch == nil {
+		c.TimeStampEpoch = cfg.TimeStampEpoch
+	}
+	if c.ExpandEventListFromField == "" {
+		c.ExpandEventListFromField = cfg.ExpandEventListFromField
+	}
+	if len(c.FileSelectors) == 0 && len(cfg.FileSelectors) > 0 {
+		c.FileSelectors = append(c.FileSelectors, cfg.FileSelectors...)
+	}
 	return c
 }
 
+// isValidUnixTimestamp checks if the timestamp is a valid Unix timestamp
+func isValidUnixTimestamp(timestamp int64) bool {
+	// defines the valid range for Unix timestamps
+	minTimestamp := time.Date(1970, time.January, 1, 0, 0, 0, 0, time.UTC).Unix()
+	maxTimestamp := time.Date(2106, time.February, 7, 6, 28, 15, 0, time.UTC).Unix()
```

Comment:
The date "2106-02-07" is significant in the context of Unix timestamps because it represents the maximum representable date for a 32-bit Unix timestamp. But since we are using 64-bit integers here, i'll update it to ```maxTimestamp := time.Date(3000, time.January, 1, 0, 0, 0, 0, time.UTC).Unix()```

Reply:
Ahh... `time_t` is specified weakly and I would say is probably most commonly defined from an `int`. So the date that is commonly remarked as the end of time is 2038-01-19.